### PR TITLE
Hotfix animal emblem ranks

### PR DESCRIPTION
**Pull request by LeroyAndrade**
+ animal emblems shown
+ animal title shown

### CODE REVIEW

File: src/system/utility.tsx [hunk 1]
```diff
@@ -139,6 +146,63 @@ export const getRankId = (name: string) => {
...
+
+
+
+
+
+
+
+// PlusMinus temporary hotfix
+//below
+export const arrayAnimalRanks = [
```
Comment (cipherxof) [inline src/system/utility.tsx]: We don't really need a lookup table for the rank images. I'll just store the ranks on the server using the ID instead.

File: src/pages/RankGuide.tsx [hunk 2]
```diff
@@ -469,5 +507,32 @@ export default (): React.ReactNode => {
     <PageContainer>
       <div className="row">{cards}</div>
     </PageContainer>
-  );
-};
+  )};
+
+// for (const rank of ranks) {
```
Comment (cipherxof) [inline src/pages/RankGuide.tsx]: Remove all of the unused code.

File: src/pages/RankGuide.tsx [hunk 3]
```diff
@@ -449,16 +483,20 @@ export default (): React.ReactNode => {
...
       ),
-    },
+    }
   ];
-
   const cards: JSX.Element[] = [];
 
-  for (const rank of ranks) {
+  let countArrayAnimalRanks=1;
+  const pathToImg = `https://mgo2pc.com/static/media/emblem/${arrayAnimalRanks[countArrayAnimalRanks][1]}.png`;
```
Comment (cipherxof) [inline src/pages/RankGuide.tsx]: This variable is unused.

File: src/pages/RankGuide.tsx [hunk 4]
```diff
@@ -1,20 +1,26 @@
-import { getRankId, getRankPreview } from '@/system/utility';
+import { getRankId, getRankPreview, getRankImage, arrayAnimalRanks } from '@/system/utility';
 import { PageContainer } from '@ant-design/pro-layout';
-import { Card, Typography } from 'antd';
+import { Card, Descriptions, Typography } from 'antd';
```
Comment (cipherxof) [inline src/pages/RankGuide.tsx:3]: Remove unused imports.

File: src/pages/RankGuide.tsx [hunk 5]
```diff
@@ -1,20 +1,26 @@
...
 import React from 'react';
+import {useState}  from 'react';
```
Comment (cipherxof) [inline src/pages/RankGuide.tsx]: unused

File: src/system/utility.tsx [hunk 6]
```diff
@@ -68,7 +68,14 @@ export const logoutUser = () => {
 };
 
 export const getRankPreview = (rank: number) => {
-  return `/static/media/emblem/${rank}.png`;
+  return `https://mgo2pc.com/static/media/emblem/${rank}.png`;
```
Comment (cipherxof) [inline src/system/utility.tsx]: should be using relative paths

**Review by cipherxof — CHANGES_REQUESTED** [review]
In general, I would remove all of your changes except for where you added the ID to the rank images. From there, I'll store the images on the server using the ID which is the simplest solution.

**Comment by LeroyAndrade** [comment]
Thank you for the feedback.
As of now only the variable containing the absolute path should be edited to the relative path (RankGuide.tsx -> variable: pathToImgAndShowRankEmblem on line #493.
Other then that, all of your requests have been honored and executed.

**Comment by cipherxof** [comment]
Seems good 👍

**Pull request merged**